repos/system_upgrade/el7toel8/libraries/satelliteutils.py

```diff
@@ -0,0 +1,23 @@
+from leapp.exceptions import StopActorExecutionError
+from leapp.libraries.common import mounting
+from leapp.libraries.stdlib import api, CalledProcessError
+
+
+def apply_postgresql_workaround(context=None):
+    """
+    Applies a workaround on the system to allow the upgrade to succeed for PostgreSQL.
+    """
+    script_path = api.get_tool_path('handle-postgresql-legacy-actions')
+    cmd = ['/bin/bash', '-c', script_path]
```

Comment:
One of the things I don't like about this, is that this is now executed always. I am still thinking of how to best mitigate this. It should be only executed if satellite is installed. That's one of the 'issues' with this workaround.
You're now also executing the the removal of that folder if postgresql (kind of any that has a symlink there) is installed even without any satellite around. 
I have to think of something.

Reply:
Just be sure, there's nothing wrong with your implementation - I am just trying to find a good way to not execute it if there's no satellite. Without having to consume the satellite message everywhere.